godtools/App/Share/Domain/UseCases/GetDeviceLanguageCodeUseCase.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  GetDeviceLanguageCodeUseCase.swift
+//  godtools
+//
+//  Created by Rachael Skeath on 7/25/22.
+//  Copyright © 2022 Cru. All rights reserved.
+//
+
+import Foundation
+
+class GetDeviceLanguageCodeUseCase {
+    
+    private let deviceLanguage: DeviceLanguageType
+    
+    init(deviceLanguage: DeviceLanguageType) {
+        
+        self.deviceLanguage = deviceLanguage
+    }
+    
+    func getDeviceLanguageCode() -> String {
```

Comment:
I have some additional thoughts, instead of this UseCase being specific to a LanguageCode, how about having a general ```GetDeviceLanguageUseCase```.

For the return type instead of a ```String```, a ```DeviceLanguageDomainModel``` which can encapsulate the languageCode.  This would give us some room to attach additional device language related attributes to our domain model if needed.

Thoughts?

Reply:
That sounds good!  That'll make it easy to build on in the future.  I'm wary of having millions of tiny useCases that do almost nothing, so grouping things together when reasonable will help that.

Reply:
Yeah I am wary of that as well.  It's something we will most likely have to balance and there may be situations where it's okay not to use a UseCase and directly inject data dependencies into the ViewModel.  We'll have explore that as we go.